File: backends/php-laravel/app/Domain/Staff/Condition/StaffCondition.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\Domain\Staff\Condition;

use App\Domain\Staff\Enums\Provider;
use App\Domain\Staff\Enums\StaffRole;
use App\Domain\Staff\Enums\StaffStatus;
use App\Support\Repositories\Conditions\AbstractCondition;
use Carbon\Carbon;

/**
 * スタッフConditionクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Staff\Condition
 */
class StaffCondition extends AbstractCondition
{
    public ?int $id = null;
    public ?string $name = null;
    public ?string $email = null;
    public ?Provider $provider = null;
    public ?string $providerId = null;
    public ?string $avater = null;
    public ?StaffRole $role = null;
    public ?StaffStatus $status = null;
    public ?Carbon $lastLoginAt = null;

    /**
     * 一覧検索用キーワード（名前・識別子の部分一致）。API の keyword からマッピングします。
     */
    public ?string $keyword = null;

    /**
     * 権限コードの一覧（空は無条件）
     *
     * @var array<int, int>
     */
    public array $roles = [];

    /**
     * 状態コードの一覧（空は無条件）
     *
     * @var array<int, int>
     */
    public array $statuses = [];
}

File: backends/java-spring/src/main/java/com/authorization/domain/staff/condition/StaffCondition.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.staff.condition;

import com.authorization.domain.staff.enums.Provider;
import com.authorization.domain.staff.enums.StaffRole;
import com.authorization.domain.staff.enums.StaffStatus;
import com.authorization.support.repositories.conditions.AbstractCondition;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * スタッフConditionクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class StaffCondition extends AbstractCondition {

    private String name;
    private String email;
    private Provider provider;
    private String providerId;
    private String avatar;
    private StaffRole role;
    private StaffStatus status;
    private LocalDateTime lastLoginAt;

    /** 一覧検索用キーワード（名前・メールアドレスの部分一致）。 */
    private String keyword;

    /** 権限コードの一覧（空は無条件）。 */
    private List<Integer> roles = List.of();

    /** 状態コードの一覧（空は無条件）。 */
    private List<Integer> statuses = List.of();
}
